src/pybind/rbd/rbd.pyx

```diff
@@ -2553,3 +2647,54 @@ cdef class SnapIterator(object):
         if self.snaps:
             rbd_snap_list_end(self.snaps)
             free(self.snaps)
+
+cdef class TrashIterator(object):
+    """
+    Iterator over trash entries.
+
+    Yields a dictionary containing trash info of an image.
+
+    Keys are:
+
+        * `id` (str) - image id
+
+        * `name` (str) - image name
+
+        * `source` (rbd_trash_image_source_t) - source of deletion
+
+        * `deletion_time` (time_t) - time of deletion
+
+        * `deferment_end_time` (time_t) - time that an image is allowed to be
+                                          removed from trash
+    """
+
+    cdef:
+        rados_ioctx_t ioctx
+        size_t num_entries
+        rbd_trash_image_info_t *entries
+
+    def __init__(self, ioctx):
+        self.ioctx = convert_ioctx(ioctx)
+        self.num_entries = 1024
+        self.entries = <rbd_trash_image_info_t *>realloc_chk(NULL,
+            sizeof(rbd_trash_image_info_t) * self.num_entries)
+        with nogil:
+            ret = rbd_trash_list(self.ioctx, self.entries, &self.num_entries)
+        if ret < 0:
+            raise make_ex(ret, 'error listing trash entries')
+
+    def __iter__(self):
+        for i in range(self.num_entries):
+            yield {
+                'id'          : decode_cstr(self.entries[i].id),
+                'name'        : decode_cstr(self.entries[i].name),
+                'source'      : self.entries[i].source,
```

Comment:
Convert to human-readable string -- you aren't expose the enum to the Python API